### ISSUE TITLE
Missing translations after 3.21.0 upgrade

### ISSUE THREAD

**Issue by davidromani** [sonata-project/SonataAdminBundle#4603]
#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.21.0 3.21.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.3.2  3.3.2  Symfony SonataBlockBundle
sonata-project/cache                     1.0.7  1.0.7  Cache library
sonata-project/core-bundle               3.4.0  3.4.0  Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.2.1  2.2.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2  1.0.2  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.1.6  3.1.6  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.2.0  2.2.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.7.1  1.7.1  Lightweight Exporter library
sonata-project/google-authenticator      2.0.0  2.0.0  Library to integrate Google Authenticator into a PHP project
sonata-project/user-bundle               3.2.4  3.2.4  Symfony SonataUserBundle
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
symfony/assetic-bundle     v2.8.2  v2.8.2 Integrates Assetic into Symfony2
symfony/monolog-bundle     v2.12.1 v3.1.0 Symfony MonologBundle
symfony/polyfill-apcu      v1.5.0  v1.5.0 Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu  v1.5.0  v1.5.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.5.0  v1.5.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php54     v1.5.0  v1.5.0 Symfony polyfill backporting some PHP 5.4+ features to lower PHP versions
symfony/polyfill-php55     v1.5.0  v1.5.0 Symfony polyfill backporting some PHP 5.5+ features to lower PHP versions
symfony/polyfill-php56     v1.5.0  v1.5.0 Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.5.0  v1.5.0 Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-util      v1.5.0  v1.5.0 Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.0  v3.0.0 Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v2.6.3  v3.0.3 Symfony SwiftmailerBundle
symfony/symfony            v2.8.26 v3.3.6 The Symfony PHP framework
```

#### PHP version

```
$ php -v
PHP 7.1.1 (cli) (built: Feb 13 2017 10:05:49) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.1, Copyright (c) 1999-2017, by Zend Technologies
    with Xdebug v2.5.0, Copyright (c) 2002-2016, by Derick Rethans
```

## Subject

After updating sonata-project/admin-bundle from v3.17.0 to v3.21.0 I missed all of my custom translations. Before upgrade it, same translations were working well.

## Steps to reproduce

```yaml
# app/Resources/translations/messages.ca.yml
backend:
    admin:
        contact:
            checked: Llegit
```

```php
// src/AppBundle/Admin/MyAdmin.php
    /**
     * @param ListMapper $listMapper
     */
    protected function configureListFields(ListMapper $listMapper)
    {
        $listMapper
            ->add(
                'checked',
                null,
                array(
                    'label' => 'backend.admin.contact.checked',
                )
            );
    }
```

## Expected results

A translated label with text "Llegit" in header table (list view).

## Actual results

An untranslated label with text "Checked" in header table (list view).

![captura de pantalla 2017-08-16 a les 17 42 51](https://user-images.githubusercontent.com/698779/29372156-9e8b85b2-82aa-11e7-9fa8-5a8a8e83609f.png)
![captura de pantalla 2017-08-16 a les 17 43 57](https://user-images.githubusercontent.com/698779/29372168-a43b45f6-82aa-11e7-805c-149d3a3fde99.png)

**Comment by virtualize** [comment]
I can verify this. Setting labels in **List View** is not possible anymore since v3.21.0. I downgraded to v3.20.1 and there its works flawlessly.

**Comment by jlamur** [comment]
Hello @davidromani @virtualize,

Thank you for your feedback.

This is a known bug that was introduced in `SonataAdminBundle v.3.21.0`.
It has been fixed [here](https://github.com/sonata-project/SonataAdminBundle/pull/4602) but it is not released yet.

As suggested by @virtualize you can downgrade to `v.3.20.1` and wait for the next release to be published.

**Comment by davidromani** [comment]
thanks @jlamur

**Comment by OskarStark** [comment]
I asked for a release internally

**Comment by davidromani** [comment]
@OskarStark It would be great a new release, I've several projects in production with same problem and it's quite cumbersome to manage and to keep same version in all of them.

**Comment by jlamur** [comment]
@davidromani if you add your `composer.lock` to your version control system, you probably won't have this problem.

**Comment by jlamur** [comment]
@davidromani The fix is released: [3.22.0](https://github.com/sonata-project/SonataAdminBundle/releases/tag/3.22.0)
